client/src/app/shared/directives/listen-editing.directive.ts

```diff
@@ -0,0 +1,177 @@
+        this.isEditing = isSo;
+        if (isSo) {
+            this.enterEditMode();
+        } else {
+            this.leaveEditMode();
+        }
+    }
+
+    public get osListenEditing(): boolean {
+        return this.isEditing;
+    }
+
+    @Input()
+    public baseModel: BaseModel;
+
+    /**
+     * Array to recognize, if there are other persons working on the same
+     * base-model and see, if those persons leave the editing-view.
+     */
+    private otherWorkOnBaseModel = [];
+
+    /**
+     * The variable to hold the subscription for notifications in editing-view.
+     * Necessary to unsubscribe after leaving the editing-view.
+     */
+    private editNotificationSubscription: Subscription;
+
+    /**
+     * Constant to identify the notification-message.
+     */
+    private EDIT_NOTIFICATION_NAME = 'editNotificationName';
+
+    private isEditing = false;
+
+    public constructor(
+        componentServiceCollector: ComponentServiceCollector,
+        private notifyService: NotifyService,
+        private operator: OperatorService
+    ) {
+        super(componentServiceCollector);
+    }
+
+    public ngOnDestroy(): void {
+        super.ngOnDestroy();
+        this.unsubscribeEditNotifications(EditNotificationType.TYPE_CLOSING_EDITING);
+    }
+
+    private enterEditMode(): void {
+        if (!this.editNotificationSubscription) {
+            this.editNotificationSubscription = this.listenToEditNotification();
+            this.sendEditNotification(EditNotificationType.TYPE_BEGIN_EDITING);
+        }
+    }
+
+    private leaveEditMode(): void {
+        this.unsubscribeEditNotifications(EditNotificationType.TYPE_CLOSING_EDITING);
+    }
+
+    /**
+     * Function to send a notification, so that other persons can recognize editing the same motion,
+     * if they're doing.
+     *
+     * @param type TypeOfNotificationViewMotion defines the type of the notification which is sent.
+     * @param user Optional userId. If set the function will send a notification to the given userId.
+     */
+    private sendEditNotification(type: EditNotificationType, user?: number): void {
+        const content: EditNotification = {
+            baseModelFqid: this.baseModel.fqid,
+            senderId: this.operator.operatorId,
+            senderName: this.operator.shortName,
+            type: type
+        };
+        if (user) {
+            this.notifyService.sendToUsers(this.EDIT_NOTIFICATION_NAME, content, user);
+        } else {
+            this.notifyService.sendToAllUsers<EditNotification>(this.EDIT_NOTIFICATION_NAME, content);
+        }
+    }
+
+    /**
+     * Function to listen to notifications if the user edits this motion.
+     * Handles the notification messages.
+     *
+     * @returns A subscription, only if the user wants to edit this motion, to listen to notifications.
+     */
+    private listenToEditNotification(): Subscription {
+        return this.notifyService
```

Comment:
This does not work currently, does it? I mean OS4 do not have this service :D

Reply:
No :D